Publish Presences - NiceHash & BetterHash

rhys19 opened the pull request:
Hello, I have made some simple presences this is my first time, I have created Nicehash & Betterhash presences.

none of these extensions release PRIVATE information they only say what the page name is your viewing and a small description of it ex: Viewing exchange prices.

I hope this gets published as I think there's a few people out there that would use these :)

Please let me know if there is anything to change on it and I gladly will!

rhys19 committed:
Add BetterHash

rhys19 committed:
Add NiceHash

rhys19 committed:
Update metadata.json

rhys19 committed:
Update metadata.json

rhys19 committed:
Update presence.ts

rhys19 committed:
Update presence.ts

ImRodry commented:
Idk how you did that but don’t upload the presence.js file

EncryptedDev commented:
You also have a few too many `metadata.json`s

rhys19 commented:
oh I didn't know, there was no instructions I could find on submitting pr so just the presence.ts then? as for the manifest do I'm not sure where to put it. do I put in /dist or the main folder

rhys19 committed:
Delete presence.js

rhys19 committed:
Delete presence.js

rhys19 committed:
Delete metadata.json

rhys19 committed:
Delete metadata.json

ImRodry commented:
There are [docs](https://docs.premid.app/en/dev/presence) on how to create a presence and git itself shouldn't ever allow you to publish the .js file, which is why I'm surprised as to how you did that, but please delete multiple files in one commit

rhys19 commented:
Ok. sorry, I deleted them via github not via local files so that's why it deleted it in multiples. so I deleted the manifest.json in the main and kept it in dist and deleted the .js is there anything else?

EncryptedDev commented on websites/B/BetterHash/presence.ts:
```diff
@@ -0,0 +1,49 @@
+      play: "presence.playback.playing",
+      pause: "presence.playback.paused"
+    }),
+  time = Math.floor(Date.now() / 1000);
```
```suggestion
  browsingTimestamp = Math.floor(Date.now() / 1000);
```

EncryptedDev commented on websites/B/BetterHash/presence.ts:
```diff
@@ -0,0 +1,49 @@
+  presence.on("UpdateData", async () => {
+    const presenceData: PresenceData = {
+    largeImageKey: "betterhash_1024x1024",
+    startTimestamp: time
```
```suggestion
    startTimestamp: browsingTimestamp
```

EncryptedDev commented on websites/N/NiceHash/dist/metadata.json:
```diff
@@ -0,0 +1,21 @@
+    "en": "Nicehash is a crypto currency miner & a hashpower marketplace!"
+  },
+  "url": "www.nicehash.com",
+  "version": "1.0.3",
```
```suggestion
  "version": "1.0.0",
```

EncryptedDev commented on websites/N/NiceHash/presence.ts:
```diff
@@ -0,0 +1,33 @@
+      play: "presence.playback.playing",
+      pause: "presence.playback.paused"
+    }),
+  time = Math.floor(Date.now() / 1000);
```
```suggestion
  browsingTImestamp = Math.floor(Date.now() / 1000);
```

EncryptedDev commented on websites/N/NiceHash/presence.ts:
```diff
@@ -0,0 +1,33 @@
+  presence.on("UpdateData", async () => {
+    const presenceData: PresenceData = {
+    largeImageKey: "nicehash_1024x1024",
+    startTimestamp: time
```
```suggestion
    startTimestamp: browsingTImestamp
```

rhys19 committed:
Update websites/N/NiceHash/presence.ts

Co-authored-by: Rhys <64903135+EncryptedDev@users.noreply.github.com>

rhys19 committed:
Update websites/N/NiceHash/presence.ts

Co-authored-by: Rhys <64903135+EncryptedDev@users.noreply.github.com>

rhys19 committed:
Update websites/N/NiceHash/dist/metadata.json

Co-authored-by: Rhys <64903135+EncryptedDev@users.noreply.github.com>

rhys19 committed:
Update websites/B/BetterHash/presence.ts

Co-authored-by: Rhys <64903135+EncryptedDev@users.noreply.github.com>

rhys19 committed:
Update websites/B/BetterHash/presence.ts

Co-authored-by: Rhys <64903135+EncryptedDev@users.noreply.github.com>

rhys19 committed:
Update websites/B/BetterHash/dist/metadata.json

Co-authored-by: Rhys <64903135+EncryptedDev@users.noreply.github.com>

ImRodry commented on websites/B/BetterHash/presence.ts:
```diff
@@ -0,0 +1,49 @@
+    }),
+  browsingTimestamp = Math.floor(Date.now() / 1000);
+
+  presence.on("UpdateData", async () => {
```
fix indentation (run prettier + eslint)

ImRodry commented on websites/N/NiceHash/presence.ts:
```diff
@@ -0,0 +1,33 @@
+    }),
+  browsingTImestamp = Math.floor(Date.now() / 1000);
+
+  presence.on("UpdateData", async () => {
```
ditto

ImRodry commented on websites/N/NiceHash/presence.ts:
```diff
@@ -0,0 +1,33 @@
+  presence.on("UpdateData", async () => {
+    const presenceData: PresenceData = {
+    largeImageKey: "nicehash_1024x1024",
+    startTimestamp: browsingTImestamp
```
that's not how you camelcase
```suggestion
    startTimestamp: browsingTimestamp
```

ImRodry commented on websites/N/NiceHash/presence.ts:
```diff
@@ -0,0 +1,33 @@
+      play: "presence.playback.playing",
+      pause: "presence.playback.paused"
+    }),
+  browsingTImestamp = Math.floor(Date.now() / 1000);
```
```suggestion
  browsingTimestamp = Math.floor(Date.now() / 1000);
```

ImRodry reviewed (changes requested):
Please batch these suggestions into one commit and use vscode, not github

rhys19 commented:
> Please batch these suggestions into one commit and use vscode, not github

ok, sorry my internet was out for like 2 hours doing it via vscode now and then uploading with the git client.

rhys19 commented:
I tried using eslint & prettier but in vscode it doesn't do any changes. it had me remove variable that wasn't used but that was it, I'm fairly new to typescript I've only coded javascript and other languages but this is my first for typescript

Commit:
Update with prettier & ESLint changes

rhys19 commented:
ESlint & Prettier did do some indentation and spaces I uploaded them via git (Used VSCode).

Bas950 commented on package.json:
```diff
@@ -29,8 +29,8 @@
     "@types/semver": "^7.3.8",
-    "@typescript-eslint/eslint-plugin": "^4.33.0",
-    "@typescript-eslint/parser": "^4.33.0",
+    "@typescript-eslint/eslint-plugin": "^5.6.0",
+    "@typescript-eslint/parser": "^5.0.0",
```
undo

rhys19 replied:
fixed

Commit:
Undo parser and eslint version update

EncryptedDev commented:
Fix this PR and reopen

rhys19 commented:
> Fix this PR and reopen

my question is how come the coinbase extension was merged, when mine isn't that different???

it's my first time doing typescript I don't understand what's wrong with it...
been trying to understand what was wrong but if someone could just point me in the right direction it'd be nice.

EncryptedDev commented:
Only commit the files you want to change, nothing more. You have commited random files such as: `package.json.bak` and deleted `yarn.lock`
